File: Tech-Module/Software_Technologies/09. Java-Syntax-And-Basics/src/exercises/EqualSums.java
package exercises;

import java.util.Arrays;
import java.util.Scanner;

public class EqualSums {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int[] numbers = Arrays.stream(scanner.nextLine().split("\\s+")).mapToInt(Integer::parseInt).toArray();
        int leftSum = 0;
        int rightSum = 0;
        int count = 0;

        for (int i = 0; i < numbers.length; i++) {
            int currentNum = numbers[i];

            for (int j = 0; j < i; j++) {
                leftSum += numbers[j];
            }
            for (int j = i + 1; j < numbers.length; j++) {
                rightSum += numbers[j];
            }

            if (leftSum == rightSum) {
                System.out.println(i);
                count++;
                break;
            } else {
                leftSum = 0;
                rightSum = 0;
            }
        }
        if (count == 0) {
            System.out.println("no");
        }
    }
}

File: Tech-Module/Programming_Fundametals/06_Arrays/Exercises/11_Equal_Sums/EqualSums.cs
using System.Linq;

namespace _11_Equal_Sums
{
    using System;

    public class EqualSums
    {
        public static void Main()
        {
            var input = Console.ReadLine()
                .Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries)
                .Select(int.Parse)
                .ToArray();

            var count = 0;

            for (var i = 0; i < input.Length; i++)
            {
                var leftSum = 0;
                var rightSum = 0;

                if (i == 0)
                {
                    leftSum = 0;

                    for (var j = 1; j < input.Length; j++)
                    {
                        rightSum += input[j];
                    }
                }
                else
                {
                    for (var j = 0; j < i; j++)
                    {
                        leftSum += input[j];
                    }

                    for (var k = i + 1; k < input.Length; k++)
                    {
                        rightSum += input[k];
                    }
                }

                if (leftSum == rightSum)
                {
                    Console.WriteLine(i);
                    count++;
                }
            }

            if (count == 0)
            {
                Console.WriteLine("no");
            }
        }
    }
}